generator/lookml.py

```diff
@@ -0,0 +1,148 @@
+"""Generate lookml from namespaces."""
+import re
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+import click
+import yaml
+from google.cloud import bigquery
+from jinja2 import Environment, FileSystemLoader
+
+_env = Environment(
+    loader=FileSystemLoader(
+        (Path(__file__).parent / "templates").absolute().as_posix()
+    ),
+    trim_blocks=True,
+    lstrip_blocks=True,
+)
+
+VIEW_TEMPLATE = _env.get_template("view.lkml")
+
+
+@dataclass
+class Dimension:
+    """LookML View Dimension."""
+
+    path: Tuple[str, ...]
+    field_type: str
+    mode: str
+
+    @property
+    def is_group(self) -> bool:
+        """Determine if this is a dimension_group or dimension."""
+        return not self.hidden and self.type == "time"
+
+    @property
+    def name(self) -> str:
+        """Translate path to a dimension name."""
+        path = self.path
+        # remove suffix for dimension_group with timeframes
+        if self.field_type == "TIMESTAMP":
+            path = *path[:-1], re.sub("_(time|timestamp)$", "", self.path[-1])
+        elif self.field_type == "DATE":
+            path = *path[:-1], re.sub("_date$", "", self.path[-1])
+        # restore suffix if it was the entire pre-parsed field name
+        if path[-1] == "parsed":
+            # undo the above
+            path = self.path
+        return "__".join(path)
+
+    @property
+    def sql(self) -> str:
+        """Translate path to sql."""
+        return "${TABLE}." + ".".join(self.path)
+
+    @property
+    def hidden(self) -> bool:
+        """Determine if this is hidden."""
+        return self.mode == "REPEATED" or self.path == ("document_id",)
+
+    @property
+    def type(self) -> str:
+        """Translate bigquery field_type to lookml type."""
+        if self.field_type == "STRING":
+            return "string"
+        if self.field_type == "BOOLEAN":
+            return "yesno"
+        if self.field_type in {"INTEGER", "FLOAT"}:
+            return "number"
+        if self.field_type in {"TIMESTAMP", "DATE"}:
+            return "time"
+        raise ValueError(
+            f"field type could not be converted to dimension type {self.field_type}"
+        )
+
+    @property
+    def is_date(self) -> bool:
+        """Determine if this is a date."""
+        return self.field_type == "DATE"
+
+    @property
+    def is_time(self) -> bool:
+        """Determine if this is a time."""
+        return self.field_type == "TIMESTAMP"
+
+    @property
+    def map_layer_name(self) -> Optional[str]:
+        """Determine map layer name."""
+        if self.path in {("country",), ("metadata", "geo", "country")}:
+            return "countries"
+        return None
+
+    @property
+    def group_label(self) -> Optional[str]:
+        """Determine group label."""
+        if len(self.path) > 1:
+            return " ".join(self.path[:-1]).replace("_", " ").title()
+        return None
```

Comment:
How is this handled in the jinja template?

Reply:
```jinja
      {% if dimension.group_label != None %}
    group_label: {{ dimension.group_label }}
    group_item_label: {{ dimension.group_item_label }}
      {% endif %}
```